yt_dlp/extractor/servus.py

```diff
@@ -15,130 +12,97 @@ class ServusIE(InfoExtractor):
-            elif extension == 'mp4' or ext == 'mp4':
-                formats.append({
-                    'url': format_url,
-                    'format_id': type_,
-                    'width': int_or_none(resource.get('width')),
-                    'height': int_or_none(resource.get('height')),
-                })
+            'https://api-player.redbull.com/stv/servus-tv?timeZone=Europe/Berlin&videoId=%s' % video_id,
+            video_id, 'Downloading video JSON')
+        if 'videoUrl' not in video:
+            self._report_errors(video)
+            return None
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+            video['videoUrl'], video_id, 'mp4', entry_protocol='m3u8_native',
+            m3u8_id='hls')
 
-        attrs = {}
-        for attribute in video['attributes']:
-            if not isinstance(attribute, dict):
-                continue
-            key = attribute.get('fieldKey')
-            value = attribute.get('fieldValue')
-            if not key or not value:
-                continue
-            attrs[key] = value
-
-        title = attrs.get('title_stv') or video_id
-        alt_title = attrs.get('title')
-        description = attrs.get('long_description') or attrs.get('short_description')
-        series = attrs.get('label')
-        season = attrs.get('season')
-        episode = attrs.get('chapter')
-        duration = float_or_none(attrs.get('duration'), scale=1000)
+        season = video.get('season')
         season_number = int_or_none(self._search_regex(
             r'Season (\d+)', season or '', 'season number', default=None))
+        episode = video.get('chapter')
         episode_number = int_or_none(self._search_regex(
             r'Episode (\d+)', episode or '', 'episode number', default=None))
 
         return {
             'id': video_id,
-            'title': title,
-            'alt_title': alt_title,
-            'description': description,
-            'thumbnail': thumbnail,
-            'duration': duration,
-            'timestamp': unified_timestamp(video.get('lastPublished')),
-            'series': series,
-            'season': season,
+            'title': video.get('title'),
+            'description': self._get_description(video, video_id),
+            'thumbnail': video.get('poster'),
+            'duration': float_or_none(video.get('duration')),
+            'timestamp': unified_timestamp(video.get('currentSunrise')),
+            'series': season,
+            'season': video.get('season'),
             'season_number': season_number,
             'episode': episode,
             'episode_number': episode_number,
             'formats': formats,
+            'subtitles': subtitles,
         }
+
+    def _get_description(self, video, video_id):
+        info = self._download_json("https://backend.servustv.com/wp-json/rbmh/v2/media_asset/aa_id/%s?fieldset=page" % video_id,
+                                   video_id, fatal=False)
```

Comment:
> [pukkandan](https://github.com/pukkandan) [on Jul 10, 2022](https://github.com/yt-dlp/yt-dlp/pull/4240#discussion_r917393564)
I don't think a description is useful enough to make an additional API call for it. But I leave it to your discretion

Reply:
I do not mind leaving it in, since (at least in my understanding) even if the endpoint is removed we still fallback gracefully to video.description

Reply:
How does this description from that endpoint differ from the one in `video.description`, short vs. long?

Reply:
In the test case:

```
video.get('description')
'Klettersteige in den Alpen - Vom Machtinstrument zur Trendsportart'
```

```
short description
Keine alpine Disziplin hat sich in den letzten 20 Jahren derart rasant entwickelt wie das Klettersteig-Gehen. Die Vie Ferrate, die eisernen Wege, haben ein neues Publikum auf die Berge gebracht. Es gibt Steige für Kinder, felsige Hochseilgärten oder extrem anspruchsvolle Sportklettersteige. Die Vie Ferrate sind heute ein fixer Bestandteil der Bergwelt. Wer sich ihnen mit Eigenverantwortung und einer Portion Neugier nähert, dem eröffnen die eisernen Wege faszinierende alpine Abenteuer.
```

```
long description
Weit mehr als 700 Klettersteige gibt es in ganz Österreich. Jedes Bundesland hat zumindest einen zu bieten. Jeder dieser Steige erzeugt einzigartige Emotionen und ermöglicht durchschnittlich sportlichen Menschen den Zustieg in eine alpine Sphäre, die bisher nur wagemutigen Kletterern und Spitzenbergsteigern vorbehalten war.

Der erste „Klettersteig“ entsteht in Frankreich. Der Mont Aiguille nahe Grenoble gilt bis 1492 als unbesteigbar. König Karl VIII. will ihn dennoch unterwerfen. Nach zweijähriger Planungsarbeit hämmert und nagelt sich ein königlicher Trupp mit Leitern die senkrechten Wände empor. 

350 Jahre später entsteht der erste touristische Klettersteig Europas. Der Alpenforscher Friedrich Simony sammelt beim K&amp;K-Hochadel Geld, um dieses teure Projekt am Dachstein umzusetzen. Simony will einfach andere Menschen an seiner Berg-Faszination teilhaben lassen. 

An der nächsten Generation der Klettersteige klebt Blut. Die Kriegssteige der Dolomiten wurden angelegt, um Soldaten in Stellung zu bringen oder den Nachschub zu garantieren.
```

Jahrzehnte später erscheinen die Klettersteige in einem gänzlich neuen Licht. Heli Putz ist ein österreichischer Klettersteig-Pionier. Er hat Anfang der 1990er seine ersten eigenen Steige gebaut. Als Kletterer, Bergführer und Handwerker bringt er alle Voraussetzungen dafür mit. Weit mehr als 30 Routen hat er bis heute eingebohrt. Einige, wie der Steig am Donnerkogel am Gosaukamm, haben Kultstatus erreicht und wurden zu Instagram-Hotspots. Genau diesen Steig saniert Heli nach 20 Jahren und verpasst der Tour im unteren Bereich eine neue Routenführung.

Wie gut ihm diese neue Route am Donnerkogel gelungen ist, testet Heli gemeinsam mit Spitzenbergsteiger Christian Stangl bei einer spektakulären Winterbegehung.

Reply:
Then I agree that the API call should be done, since the `video.description` is no more than a subtitle really.

Interestingly short and long description differ drastically and do not really contain the same information. I think it might make sense to join them into one?

Also apparently the descriptions need to be html-decoded, as seen in the `K&amp;K-Hochadel` part of the long description.

Reply:
Yeah makes sense, they are also merged in the servus tv UI:
![image](https://user-images.githubusercontent.com/36509607/212552119-685e928f-58c7-4dc9-8c3d-85e09237e7b7.png)

Merged and decoded result looks like this (f09d6396):
```
Keine alpine Disziplin hat sich in den letzten 20 Jahren derart rasant entwickelt wie das Klettersteig-Gehen. Die Vie Ferrate, die eisernen Wege, haben ein neues Publikum auf die Berge gebracht. Es gibt Steige für Kinder, felsige Hochseilgärten oder extrem anspruchsvolle Sportklettersteige. Die Vie Ferrate sind heute ein fixer Bestandteil der Bergwelt. Wer sich ihnen mit Eigenverantwortung und einer Portion Neugier nähert, dem eröffnen die eisernen Wege faszinierende alpine Abenteuer.


Weit mehr als 700 Klettersteige gibt es in ganz Österreich. Jedes Bundesland hat zumindest einen zu bieten. Jeder dieser Steige erzeugt einzigartige Emotionen und ermöglicht durchschnittlich sportlichen Menschen den Zustieg in eine alpine Sphäre, die bisher nur wagemutigen Kletterern und Spitzenbergsteigern vorbehalten war.

Der erste „Klettersteig“ entsteht in Frankreich. Der Mont Aiguille nahe Grenoble gilt bis 1492 als unbesteigbar. König Karl VIII. will ihn dennoch unterwerfen. Nach zweijähriger Planungsarbeit hämmert und nagelt sich ein königlicher Trupp mit Leitern die senkrechten Wände empor. 

350 Jahre später entsteht der erste touristische Klettersteig Europas. Der Alpenforscher Friedrich Simony sammelt beim K&K-Hochadel Geld, um dieses teure Projekt am Dachstein umzusetzen. Simony will einfach andere Menschen an seiner Berg-Faszination teilhaben lassen. 

An der nächsten Generation der Klettersteige klebt Blut. Die Kriegssteige der Dolomiten wurden angelegt, um Soldaten in Stellung zu bringen oder den Nachschub zu garantieren.

Jahrzehnte später erscheinen die Klettersteige in einem gänzlich neuen Licht. Heli Putz ist ein österreichischer Klettersteig-Pionier. Er hat Anfang der 1990er seine ersten eigenen Steige gebaut. Als Kletterer, Bergführer und Handwerker bringt er alle Voraussetzungen dafür mit. Weit mehr als 30 Routen hat er bis heute eingebohrt. Einige, wie der Steig am Donnerkogel am Gosaukamm, haben Kultstatus erreicht und wurden zu Instagram-Hotspots. Genau diesen Steig saniert Heli nach 20 Jahren und verpasst der Tour im unteren Bereich eine neue Routenführung.

Wie gut ihm diese neue Route am Donnerkogel gelungen ist, testet Heli gemeinsam mit Spitzenbergsteiger Christian Stangl bei einer spektakulären Winterbegehung.

```

Reply:
Would it make sense to replace `\n\n` with `\n`? Otherwise LGTM

Reply:
Actually this is intentional to slightly more resemble the Lay-out in Servus tv. I also considered merging them into one to make it even more similar, but that could be double edged. So for now 3 new lines make it easy to see where the short description ends

Reply:
Since the newlines in the long description group related sentences I think keeping them is fine, I just wanted to get rid of the doubling since that is redundant. The short and long description would still be separated by a double newline, clearly separating them:
```py
        short_description = unescapeHTML(traverse_obj(info, 'stv_short_description', expected_type=str))
        long_description = unescapeHTML(traverse_obj(info, 'stv_long_description', expected_type=str, default='')).replace('\n\n', '\n')

        if short_description and long_description:
            return f'{short_description}\n\n{long_description}'
        return long_description or short_description
```

Reply:
Ahh I misunderstood :) yeah let’s replace it then

Reply:
2a6dd3fe